### PR TITLE
enabled test interface to override default behavior of reloading targ…

### PR DESCRIPTION
**Pull request by info-rchitect**
…ets no matter what

### CODE REVIEW

**Comment by priyavadan** [comment]
@info-rchitect can you please look into why the CI builds are failing. Seems like the flow_control test flow has a comment difference

**Comment by info-rchitect** [comment]
The last commit was just code style to ensure the private method returns a Boolean in all situations

**Comment by priyavadan** [comment]
@ThaoHuynhFsl this should solve the long generation times for ASF as well, I think; do you want to trial it out please?

File: lib/origen_testers/origen_ext/generator/flow.rb [hunk 1]
```diff
@@ -106,6 +108,10 @@ def _create(options = {}, &block)
         job.output_file_body = options.delete(:name).to_s if options[:name]
         if sub_flow
           interface = Origen.interface
+          if reload_target?(interface, options)
+            Origen.app.reload_target!
+            Origen.tester.generating = :program
+          end
```
Comment (ginty) [inline lib/origen_testers/origen_ext/generator/flow.rb]: Can you comment on why the reloading has been moved from its original position?

Comment (info-rchitect) [reply lib/origen_testers/origen_ext/generator/flow.rb]: @ginty Previously the target was reloaded before the interface was instantiated so there was no chance for the interface to alter the outcome.

**Comment by info-rchitect** [comment]
> OK, makes sense, thanks for the clarification.
> I think this feature should be briefly mentioned in either the Flow or Interface guide though.

@ginty will do, one sec.

**Pull request merged**